core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -706,8 +708,18 @@ angular.module('oppia').directive('questionsList', [
           $scope.$on(EVENT_QUESTION_SUMMARIES_INITIALIZED, function(ev) {
             _initTab(false);
           });
-
-          _initTab(true);
+          ctrl.$onInit = function() {
+            ctrl.skillIds = [];
+            ctrl.selectedSkillIds = [];
+            ctrl.associatedSkillSummaries = [];
+            ctrl.selectedSkillId = ctrl.getSelectedSkillId();
+            ctrl.getQuestionSummariesForOneSkill =
+              QuestionsListService.getCachedQuestionSummaries;
```

Comment:
Again, this and the next line are functions. I'll stop commenting on these instances but once we've figured out the correct resolution to this sort of construct, let's apply it throughout the other files too.

Reply:
Updated everywhere.